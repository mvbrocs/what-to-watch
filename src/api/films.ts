import { appAxios } from './';

export interface IFilm {
  name: string;
  poster_image: string;
  preview_image: string;
  background_image: string;
  background_color: string;
  description: string;
  rating: number;
  scores_count: number;
  director: string;
  starring: string[];
  run_time: number;
  genre: string;
  released: number;
  id: number;
  is_favorite: boolean;
  video_link: string;
  preview_video_link: string;
}

export const filmsAPI = {
  getAll: () => appAxios.get<IFilm[]>('/films'),
};
