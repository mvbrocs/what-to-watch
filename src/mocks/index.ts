export { default as mockFilms } from './films';
export { default as mockStore } from './store';
