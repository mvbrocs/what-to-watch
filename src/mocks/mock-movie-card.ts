import { mockMovies } from './mock-movies';

export const mockMovieCard = mockMovies[0];
