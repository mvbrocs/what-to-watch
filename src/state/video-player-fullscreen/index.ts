import { videoPlayerFullscreenSlice } from './video-player-fullscreen-slice';

export type VideoPlayerFullscreenSliceState = {
  visible: boolean;
  background_image: string | null;
  background_color: string | null;
  name: string | null;
  video_link: string | null;
};

export { videoPlayerFullscreenInitialState } from './video-player-fullscreen-slice';
export const videoPlayerFullscreenReducer = videoPlayerFullscreenSlice.reducer;
export const {
  toggleVideoPlayerFullscreenVisible,
  updateVideoPlayerFullscreenData,
} = videoPlayerFullscreenSlice.actions;
export { selectVideoPlayerFullscreenState } from './video-player-fullscreen-selectors';
