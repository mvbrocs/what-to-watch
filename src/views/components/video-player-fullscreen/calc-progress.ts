export const calcProgress = (current: number, total: number) => (100 * current) / total;
