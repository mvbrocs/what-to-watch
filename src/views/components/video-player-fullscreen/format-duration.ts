import { MIN_IN_HR } from 'src/views/utils';

const SEC_IN_MIN = 60;

export const addZero = (num: number) => (num.toString().length === 1 ? `0${num}` : `${num}`);

export const formatDuration = (sec: number) => {
  const totalSec = Math.floor(sec);
  const totalMin = Math.floor(totalSec / SEC_IN_MIN);
  const totalHr = Math.floor(totalMin / MIN_IN_HR);

  const manySec = totalSec - totalMin * SEC_IN_MIN;
  const manyMin = totalMin - totalHr * MIN_IN_HR;

  const durationSec = addZero(manySec);
  const durationMin = addZero(manyMin);
  const durationHr = addZero(totalHr);

  return `${durationHr}:${durationMin}:${durationSec}`;
};
