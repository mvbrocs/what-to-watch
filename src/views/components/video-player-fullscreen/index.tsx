import { RefObject } from 'react';

export type VideoPlayerFullscreenPlayHandler = (videoRef: RefObject<HTMLVideoElement>) => void;
export { VideoPlayerFullscreen } from './video-player-fullscreen';
