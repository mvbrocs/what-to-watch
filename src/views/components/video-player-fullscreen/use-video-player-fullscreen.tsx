import { useRef } from 'react';

export const useVideoPlayerFullscreen = () => {
  const videoRef = useRef<HTMLVideoElement>(null);

  return { videoRef };
};
