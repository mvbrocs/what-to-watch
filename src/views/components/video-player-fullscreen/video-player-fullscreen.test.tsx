import React from 'react';
import { shallow } from 'enzyme';

import { VideoPlayerFullscreen } from './video-player-fullscreen';
import { mockFilms } from 'mocks/films';

describe('VideoPlayerFullscreen', () => {
  const closeHandlerMock = jest.fn();
  const playHandlerMock = jest.fn();
  const props = {
    visible: true,
    background_image: mockFilms[0].background_image,
    background_color: mockFilms[0].background_color,
    name: mockFilms[0].name,
    video_link: mockFilms[0].video_link,
    onClose: closeHandlerMock,
    onPlay: playHandlerMock,
  };
  const videoPlayerFullscreen = shallow(<VideoPlayerFullscreen {...props} />);

  test('renders properly', () => {
    expect(videoPlayerFullscreen).toMatchSnapshot();
  });

  test('renders nothing if visible false', () => {
    const videoPlayerFullscreenNotVisible = shallow(
      <VideoPlayerFullscreen {...props} visible={false} />,
    );

    expect(videoPlayerFullscreenNotVisible).toMatchSnapshot();
  });

  test('call close handler on btn exit click', () => {
    videoPlayerFullscreen.find('.player__exit').simulate('click');

    expect(closeHandlerMock).toBeCalledTimes(1);
  });

  test('call play handler on btn play click', () => {
    videoPlayerFullscreen.find('.player__play').simulate('click');

    expect(playHandlerMock).toBeCalledTimes(1);
  });
});
