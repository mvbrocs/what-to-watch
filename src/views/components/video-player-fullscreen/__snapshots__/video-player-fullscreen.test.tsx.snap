// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayerFullscreen renders nothing if visible false 1`] = `""`;

exports[`VideoPlayerFullscreen renders properly 1`] = `
<div
  className="player"
  style={
    Object {
      "backgroundColor": "#D8D3BD",
    }
  }
>
  <video
    className="player__video"
    poster="https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Orlando.jpg"
    src="http://media.xiph.org/mango/tears_of_steel_1080p.webm"
  />
  <button
    className="player__exit"
    onClick={[MockFunction]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value="30"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "30%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        1:30:29
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Orlando
      </div>
      <button
        className="player__full-screen"
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
